### DISCUSSION TITLE
How to access filtered data in a DataTable

### DISCUSSION THREAD

**Discussion by Mane75** [primefaces/community#252] [PrimeVue]
Hi folks,

I need to export to XLS the content of a DataTable after the users has apply some filters. Unfortunatly I haven't found a way to access the filtered data. This **this.$refs.dt.value** has the all the unlfiltered data, but is there a way to access the filtered data(to the user)?

Regards
Mario

**Comment by palonza** [comment]
Hi Mario,

In the exportToXLS method, you can use: this.$refs.dt.filteredValue will give you access to the data that has been filtered by users in your DataTable. You can use a library like xlsx to convert this data to an XLS file and provide a download option for users.

then use this.$refs.dt.filteredValue insted of this.$refs.dt.value

I hope it helps you, if so, mark the question as answered to help me.

Regards!

**Reply by Mane75** [reply]
Thank you Jose, unfortunately I tried this.$refs.dt.filteredValue but apperently it is not available. I did check the source code and it only reference if a parameter in the onfilter event.
